bundles/core/org.eclipse.smarthome.core.library/src/main/java/org/eclipse/smarthome/core/library/types/PointType.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.smarthome.core.library.types;
+
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.ComplexType;
+import org.eclipse.smarthome.core.types.PrimitiveType;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * This type can be used for items that are dealing with GPS or
+ * location awareness functionality.
+ *
+ * @author Gaël L'hopital
+ *
+ */
+public class PointType implements ComplexType, Command, State {
+
+    public static final double EARTH_GRAVITATIONAL_CONSTANT = 3.986004418e14;
+    public static final double WGS84_a = 6378137; // The equatorial radius of WGS84 ellipsoid (6378137 m).
+
+    private double latitude; // in decimal degrees
```

Comment:
Is there any reason why you internally work with double and not with BigDecimal? Using BigDecimal would make sure not to lose any precision and also not ending up with ugly rounding problems. Using the constructor should result in the constituents being an exact match to what was reached into the constructor (assuming the values were canonical).